### ISSUE TITLE
FatalErrorException in MorphTo.php line 112: Call to undefined method App\Model\Entities\Work::getEntityKey()

### ISSUE THREAD

**Issue by Evertt** [analogueorm/analogue#66]
Title says it all, I'll fix it tomorrow probably. :-)

**Comment by RemiCollin** [comment]
Can you please switch to the system-refactoring branch and tell me if you have this issue ?

**Comment by Evertt** [comment]
Nope, it seems to be fixed there. :-)

**Comment by RemiCollin** [comment]
Ok great! It's nearly done, I'll patch to the main branch soon.

**Comment by Evertt** [comment]
I look forward to it. Would you also like to give a short summary of what you changed and what consequences it has?
I can imagine summarizing so many changes might be difficult, but I thought I’d still ask, because maybe it’s not difficult. :-p

**Comment by RemiCollin** [comment]
I had to refactor some parts of the code that really needed it, especially the storing process. A lot of the logic is now into the new Aggregate class, which uses a 'tree-like' algorithm to parse entities & relationships. It makes the process much simpler and robust, and removed a lot of duplicate code.

Also internally, entities are handled using an adapter class, which will make the client-side implementation easier, and eventually allow us to use Plain PHP Object as entities, without the need to extends Entity or the Mappable Interface.  

In short, not a lot of new stuff, but the needed foundation for the next features to come.

**Comment by Evertt** [comment]
Cool, thank you!

**Comment by Evertt** [comment]
When I read "next features to come", I wonder, what do you have in mind? :-D

**Comment by RemiCollin** [comment]
Plain Old PHP Object support will come first. Not sure about the other features yet :)

**Comment by Evertt** [comment]
When are you ready with it? I'm eager to write some new pull-requests, but I'm worried it's useless if I do it before you merge your patch. :-)

**Comment by RemiCollin** [comment]
Still has some bugs to address until all tests pass. Probably by the end of today.
